File: mobile-client/android/core/push/src/main/java/com/smartresponsor/core/push/Push.kt
package com.smartresponsor.core.push

import com.smartresponsor.mobile.client.data.system.push.PushTokenRegistrar
import com.smartresponsor.mobile.client.usecase.system.push.RegisterPushTokenUseCase

/**
 * Legacy-compatible Android entry point bridged to canonical system/push slices.
 */
class Push(
    baseUrl: String = "https://httpbin.org",
) {
    private val pushTokenRegistrar: PushTokenRegistrar = PushTokenRegistrar(baseUrl)
    private val registerPushTokenUseCase: RegisterPushTokenUseCase = RegisterPushTokenUseCase(pushTokenRegistrar)

    fun register(token: String, platform: String = "android"): Boolean =
        registerPushTokenUseCase(token = token, platform = platform)
}

File: mobile-client/ios/Sources/CorePush/Push.swift
import Foundation

public struct Push {
  private let registerPushTokenUseCase: RegisterPushTokenUseCase
  private let pushTokenRegistrar: PushTokenRegistrar

  public init(
    pushTokenRegistrar: PushTokenRegistrar = PushTokenRegistrar()
  ) {
    self.pushTokenRegistrar = pushTokenRegistrar
    self.registerPushTokenUseCase = RegisterPushTokenUseCase(pushTokenRegistrar: pushTokenRegistrar)
  }

  public func register(token: String, platform: String = "ios") async throws -> Bool {
    try await registerPushTokenUseCase(token: token, platform: platform)
  }

  public func registrationPayload(token: String, platform: String = "ios") -> PushRegistrationPayload {
    PushRegistrationPayload(token: token, platform: platform)
  }
}
